src/Sylius/Bundle/UiBundle/Registry/TemplateBlock.php

```diff
@@ -22,6 +22,7 @@ public function __construct(
         private ?array $context,
         private ?int $priority,
         private ?bool $enabled,
+        private ?string $component = null,
```

Comment:
Should it throw exception if both `template` and `component` are set?
If `TemplateBlock` allows only one of then, then it shouldn't allow misconfiguration.

Reply:
To further indicate, that those are two different modes of work, we can add `TemplateBlock->getType()` which will be set based one `$template/$component` values presence